packages/devtools/lib/src/timeline/cpu_flame_chart.dart

```diff
@@ -0,0 +1,169 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'dart:async';
+import 'dart:math' as math;
+import '../ui/drag_scroll.dart';
+import '../ui/fake_flutter/dart_ui/dart_ui.dart';
+import 'cpu_profile_protocol.dart';
+import 'flame_chart.dart';
+
+final StreamController<CpuFlameChartItem> _selectedCpuFlameChartItemController =
+    StreamController<CpuFlameChartItem>.broadcast();
+
+Stream<CpuFlameChartItem> get onSelectedCpuFlameChartItem =>
+    _selectedCpuFlameChartItemController.stream;
+
+final DragScroll _dragScroll = DragScroll();
+
+// TODO(kenzie): get feedback on this palette and add dark theme colors.
+const colorPalette = [
+  Color(0xFFFF5722),
+  Color(0xFFFF6D00),
+  Color(0xFFFFAB00),
+  Color(0xFFFFD600),
+  Color(0xFFFFEA00),
+];
+
+class CpuFlameChart extends FlameChart<CpuProfileData> {
+  CpuFlameChart()
+      : super(
+          onSelectedFlameChartItem: onSelectedCpuFlameChartItem,
+          dragScroll: _dragScroll,
+          classes: 'cpu-flame-chart ui-details-section',
+          flameChartInset: 4,
+        );
+
+  static const samplePadding = 1;
+
+  int _colorOffset = 0;
+
+  // TODO(kenzie): do something more intelligent for color selection (maybe base
+  // it on cpu consumption).
+  Color nextColor() {
+    final color = colorPalette[_colorOffset % colorPalette.length];
+    _colorOffset++;
+    return color;
+  }
+
+  @override
+  void reset() {
+    super.reset();
+    _colorOffset = 0;
+  }
+
+  // TODO(kenzie): rewrite this render method to draw to canvas.
+  @override
+  void render() {
+    final CpuProfileData cpuProfileData = data;
+    final totalWidth = element.clientWidth - 2 * flameChartInset;
+
+    final Map<String, double> stackFrameLefts = {};
+
+    double calculateLeftForStackFrame(CpuStackFrame stackFrame) {
+      double left;
+      if (stackFrame.parent == null) {
+        left = flameChartInset.toDouble();
+      } else {
+        final stackFrameIndex = stackFrame.parent.children.indexOf(stackFrame);
+        if (stackFrameIndex == 0) {
+          // This is the first child of parent. [left] should equal the left
+          // value of [stackFrame]'s parent.
+          left = stackFrameLefts[stackFrame.parent.id];
+        } else {
+          // [stackFrame] is not the first child of its parent. [left] should
+          // equal the right value of its previous sibling.
+          final previous = stackFrame.parent.children[stackFrameIndex - 1];
+          left = stackFrameLefts[previous.id] +
+              (totalWidth * previous.cpuConsumptionRatio);
+        }
+      }
+      stackFrameLefts[stackFrame.id] = left;
+      return left;
+    }
+
+    void drawRecursively(CpuStackFrame stackFrame, int row) {
```

Comment:
nit: drawSubtree instead of drawRecursively to make the method name about what it does not how it does it.

Reply:
done.